src/messaging/messaging-types-internal.ts

```diff
@@ -0,0 +1,600 @@
+/*!
```

Comment:
Let's combine this into messaging-internal. We should look to keep *-internal files to a minimum.